controllers/tempo/common.go

```diff
@@ -118,3 +124,56 @@ func listFieldErrors(fieldErrs field.ErrorList) string {
 	}
 	return strings.Join(msgs, ", ")
 }
+
+func getTenantParams(
+	ctx context.Context,
+	k8sclient client.Client,
+	ctrlConfig *configv1alpha1.ProjectConfig,
+	namespace string,
+	name string,
+	tenants v1alpha1.TenantsSpec,
+	gatewayEnabled bool,
```

Comment:
wow, a lot of parameters :(

Reply:
I know, do you have a suggestion to improve it?
I think it's nice to have this one function called by `TempoStack` and `TempoMonolithic`, as both need the same logic.

Reply:
There are no too much options, I think it is fine. the other option is create an structure, but seems to be unnecessary.